### PR TITLE
Scheduler resiliency

### PR DESCRIPTION
**Pull request by cowlicks**
This makes the scheduler methods `gather`, `scatter`, and `schedule` resilient to worker failure.

### CODE REVIEW

File: dask/distributed/scheduler.py [hunk 1]
```diff
@@ -639,41 +656,44 @@ def fire_task():
                 self.trigger_task(key, dsk[key],
                         dag_state['dependencies'][key], qkey)  # Fire
 
-            try:
-                worker = self.available_workers.get(timeout=20)
-                self.available_workers.put(worker)  # put him back in
-            except Empty:
-                raise ValueError("Waited 20 seconds. No workers found")
+            with self.queue_context(qkey):
```
Comment (cowlicks) [inline dask/distributed/scheduler.py]: I could probably wrap fewer LOC with this context manager.

File: dask/distributed/tests/test_scheduler.py [hunk 2]
```diff
@@ -174,6 +175,7 @@ def test_schedule():
             else:
                 break
         assert not a.data and not b.data
+        assert len(s.queues) == 0
```
Comment (cowlicks) [inline dask/distributed/tests/test_scheduler.py]: `Scheduler.schedule` was leaving queues in `Scheduler.queues` before this.

**Comment by mrocklin** [comment]
Sorry to have been ignoring this lately

**Comment by mrocklin** [comment]
Closing this.  The existing dask.distributed scheduler is informally deprecated.  Not planning to spend more time working on it.  Hope to have something easier to work with in 3-6 months time.

**Pull request closed without merging**